### DISCUSSION TITLE
getting_started/first_2d_game/01.project_setup

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#104] [User-contributed notes]
# getting_started/first_2d_game/01.project_setup

In this short first part, we'll set up and organize the project. Launch Godot and create a new project. When creating the new project, you only need to choose a valid Project Path. You can leave th...

https://docs.godotengine.org/en/stable/getting_started/first_2d_game/01.project_setup.html

**Comment by skeebike** [comment]
The tutorial is great overall, but I am a bit confused why the default code in GDScript is static typed, but the tutorial is not static typed. When trying to use static type, the game malfunctions quite a bit and trying to debug it is quite challenging. I guess as a noob I should stick to the exact tutorial and not try to use static type, but I'm trying to establish good habits! ¯\_(ツ)_/¯

**Reply by Calinou** [reply]
> but I am a bit confused why the default code in GDScript is static typed

[This is a change that was made in 4.3.](https://github.com/godotengine/godot/pull/88026) You can disable it in the Editor Settings.

[Official demos](https://github.com/godotengine/godot-demo-projects) have also switched to static types in 4.3, but code samples in the documentation haven't been migrated over yet.

**Reply by skeebike** [reply]
This makes sense, thanks for the quick response @Calinou ! I figure it will eventually be updated in the documentation code samples. I imagine it's not exactly on top of the priority list :D

**Reply by TylerClarkTech** [reply]
If you're new, follow the tutorial and learn over time.

**Reply by lfalin** [reply]
Just a note that the editor change @Calinou mentions above turns on _type hints_, not static typing. GDScript is still dynamically typed.

**Reply by Calinou** [reply]
In GDScript, type hints are enforced at runtime (even in release builds), so it's still a form of static typing. This is different from e.g. Python where type hints are only applied by static analysis tools, not at runtime.

**Comment by BaoBaoJiu** [comment]
>This game is designed for ***portrait mode***, so we need to adjust the size of the game window. 

***portrait mode*** not in the picture.

**Reply by joyeusenoelle** [reply]
There is not a separate "portrait mode" setting. "Portrait mode" just indicates that the viewport is taller than it is wide, as opposed to "landscape mode", which is wider than it is tall. Setting the viewport to 480px wide and 720px tall makes the game be in portrait mode.

**Reply by JJC1138** [reply]
There is a project setting Display/Window/Handheld/Orientation which isn't mentioned in the tutorial (so far), but I imagine that only matters when deploying to mobile devices.

**Reply by Calinou** [reply]
Indeed, that setting has no impact on desktop platforms.

**Reply by TylerClarkTech** [reply]
portrait mode is essentially saying "make the vertical longer than the horizontal.  i.e.  make your landscape TV, vertical 9:16

**Comment by davidleejohn** [comment]
When creating a project, it would be nice to know if you are going with the "compatibility" option or one of the others. I'm sure the code works regardless, but it is supremely frustrating to have to guess and then worry the whole time that you might have goofed. If I missed something, my apologies in advance.

**Reply by jsonjunkie** [reply]
>When creating the new project, you only need to choose a valid Project Path. You can leave the other default settings alone.

for this tutorial, just set the project path and leave everything else as is

**Reply by Calinou** [reply]
The default rendering method (Forward+) was used to create the tutorial, back when Compatibility wasn't feature-complete. Now that it's feature-complete in 4.3, you can follow both the 2D and 3D *Your first game* tutorials with any rendering method.

If you use Compatibility, visuals in 3D will differ a bit due to the different lighting model (for lights that have shadows enabled), so you may have to adjust lights' energy accordingly. Visuals in 2D should be more or less identical.

**Comment by aidan-sage** [comment]
Can someone please help me?
I literally just started the project and I am getting an error at line 1 'Unexpected "Identifier" in class body.' 
The error is for the line that says "extends Area2D", which literally just comes pre-typed when you create an Area2D node. This is happening in my player script. 
Additionally, when I typed in "export (int) var speed" I cannot see anywhere to adjust the speed within the inspector panel. 
I'm using the most recent version of Godot. Does it have something to do with a recent update?
It's so frustrating because I've already completed this entire tutorial and now I can't even get past the first few steps. I have no clue why this is happening. I'm brand new to coding and stuff like this is so irritating.

**Reply by aidan-sage** [reply]
I figured it out with help from someone on reddit. For anyone else having this issue, the line for exporting the speed variable is now:
@export var speed: int
this is because of an update to Godot, I think.

**Comment by mark-wiemer** [comment]
Is my Godot weird or is the "Create" button now named "New"? If so, an updated screenshot would help beginners :)

**Reply by mark-wiemer** [reply]
Hmm, I was using Godot 4.2, not sure if it changed in 4.3, my bad

**Comment by lil-sassy** [comment]
I put together a little follow-along video. I figured this might be helpful for some people to see someone else go through this small project.

https://www.youtube.com/watch?v=hljPmVJEOkY
